asab/library/service.py

```
@@ -169,18 +169,26 @@ async def wait_for_library_ready(self, timeout: int = None):
-			LibraryNotReadyError: If the library is not ready within the timeout.
+			LibraryNotReadyError: If the library is not ready within the timeout, or if it is
+				not ready when this coroutine continues after the wait.
 		"""
 		if timeout is None:
 			timeout = self.LibraryReadyTimeout
+		if self.is_ready():
+			return
 		try:
 			await asyncio.wait_for(self.LibraryReadyEvent.wait(), timeout=timeout)
 		except asyncio.TimeoutError:
 			raise LibraryNotReadyError("Library is not ready yet.")
```

Ruff (0.15.20)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)